src/NUnitTestExtractor/NUnitTestExtractor.Tests/NUnitTestExtractorTests.cs

```diff
@@ -3,22 +3,309 @@
         {
+            using (StreamWriter writer = new StreamWriter(outputDirectory))
+            {
+                NUnitTestExtractorApp.GetTests(dlls, level, writer);
+            }
+        }
+
+        /// <summary>
+        /// Sets up and runs process which is used to read lines in the standard output
+        /// </summary>
+        /// <param name="dll">Dlls to pass to the process</param>
+        /// <param name="level"> level to pass to the process</param>
+        /// <returns></returns>
+       private Process SetUpProcess(string dll, string level)
+        {
+            Process process = new Process();
+
+            process.StartInfo.FileName = "NUnitTestExtractor.exe";
+            process.StartInfo.Arguments = "-l " + level + " " + dll;
+            process.StartInfo.UseShellExecute = false;
+            process.StartInfo.RedirectStandardOutput = true;
+
+            process.Start();
+
+            return process;
+        }
+
+        [SetUp]
+        public void CreateDirectoryAndFile()
+        {
+            int index = outputDirectory.LastIndexOf("/", StringComparison.OrdinalIgnoreCase);
+
+            string directory = "";
+
+            if (index > 0)
+            {
+                directory = outputDirectory.Substring(0, index);
+            }
+
+            if (directory != null && !Directory.Exists(directory))
+            {
+                Directory.CreateDirectory(directory);
+            }
+
+            if (!File.Exists(outputDirectory))
+            {
+                using (FileStream stream = File.Create(outputDirectory)) { }
+            }
+        } 
+
+        [TearDown]
+        public void EraseOutputFile()
+        {
+            if (File.Exists(outputDirectory))
+            {
+                File.Delete(outputDirectory);
+            }
+        }
+
+        [TestCaseSource(typeof(InvalidDataTestCases))]
+        public void WritingToFile_InvalidDll_NothingWritten(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            InvokeGetTestsUsingFileOutput(dlls, NUnitTestExtractorApp.ParseLevel(level));
+
+            using (StreamReader reader = new StreamReader(outputDirectory))
+            {
+                string lines = reader.ReadToEnd();
+
+                Assert.That(lines, Is.Empty);
+            }
+        }
+
+        [TestCaseSource(typeof(ValidDataTestCases))]
+        public void WritingToFile_ValidDllAndEveryLevel_GetsWritten(string dll, string level)
+        {
+            List<string> dlls = new List<string>();
+            dlls.Add(dll);
+
+            InvokeGetTestsUsingFileOutput(dlls, NUnitTestExtractorApp.ParseLevel(level));
+
+            using (StreamReader reader = new StreamReader(outputDirectory))
+            {
+                string lines = reader.ReadToEnd();
+
+                Assert.That(lines, Is.Not.Empty);
```

Comment:
It should probably verify more than just that it's not empty.
It might have garbage in the file and the test would still pass.  You should verify that the content is correct.